pygmt/tests/test_triangulate.py

```diff
@@ -43,7 +43,8 @@ def fixture_expected_dataframe():
             [4, 6, 1],
             [3, 4, 2],
             [9, 3, 8],
-        ]
+        ],
+        dtype=float,
```

Comment:
When writing to a file, GMT uses a format specifier like `%g` to write float-pointing data. For example, the output of the GMT CLI is:
```
> echo 1.0 1.0 | gmt select -R0/10/0/10
1   1
```
When we read this file via `pd.read_csv`, the columns will have a `int64` type. However, all data are stored as double-precision numbers in the `GMT_DATASET` object, thus column dtype will be `float64` by default. This behavior is backward incompatible, but I think the new behavior is better.

Reply:
Are there different format specifiers besides `%g`? Not too sure about storing numeric columns as float64 by default, I'd prefer a way to automatically return the lightest dtype if possible (something close to pandas' [`convert_dtypes`](https://pandas.pydata.org/pandas-docs/version/2.1/reference/api/pandas.DataFrame.convert_dtypes.html)).

Reply:
> Are there different format specifiers besides `%g`? 

The format of these outputs is controlled by GMT's `FROAMT_FLOAT_OUT` parameter and its default value is `%.12g`.

> Not too sure about storing numeric columns as float64 by default, I'd prefer a way to automatically return the lightest dtype if possible (something close to pandas' [`convert_dtypes`](https://pandas.pydata.org/pandas-docs/version/2.1/reference/api/pandas.DataFrame.convert_dtypes.html)).

Sounds good to me.